mutiny-core/src/nostr/mod.rs

```diff
@@ -1443,23 +1443,36 @@ impl<S: MutinyStorage> NostrManager<S> {
             }
         }
 
-        // handle it like a pay invoice NWC request, to see if it is valid
-        let params = PayInvoiceRequestParams {
-            id: None,
-            invoice: decrypted,
-            amount: None,
-        };
-        let invoice: Bolt11Invoice = match check_valid_nwc_invoice(&params, invoice_handler).await {
-            Ok(Some(invoice)) => invoice,
-            Ok(None) => return Ok(()),
-            Err(msg) => {
-                log_debug!(self.logger, "Not adding DM'd invoice: {msg}");
-                return Ok(());
-            }
-        };
+        // loop through dm to check for invoice
+        for word in decrypted.split_whitespace() {
+            let invoice_request_param = match bitcoin_waila::PaymentParams::from_str(word) {
+                // if not an invoice, go to next word in dm
+                Ok(param) => match param.invoice() {
+                    Some(invoice) => PayInvoiceRequestParams {
```

Comment:
I really don't like the idea of trying to parse every single word as an invoice. I think you can just look for `lnc` as a word prefix first.

Reply:
this wouldn't work with ones with the `lightning` prefix for bip21s. Maybe a better way is to filter out strings with a length less than 15

Reply:
I could ignore any string that doesn't start with an 'L' or 'b' ?

Reply:
that's still going to include a lot of words.